bfcfg

```diff
@@ -648,29 +655,157 @@ sys_cfg_one_byte()
   fi
 }
 
+#
+# Dump a multi-byte value at the specified offset in a file.
+#
+sys_cfg_multi_byte()
+{
+  local tmp_file=$1
+  local name=$2
+  local offset=$3
+  local num_bytes=$4
+  local value=$5
+
+  if [ ${dump_mode} -eq 0 ]; then
+    return
+  fi
+
+  value=0
+  if [ -e ${tmp_file} ]; then
+    value=$(hexdump -s "${offset}" -n "${num_bytes}" -e '"0x%X" "\n"' "${tmp_file}")
+  fi
+  echo "sys: ${name}=${value}"
+}
+
+#
+# This function reads the BfModes EFI struct from a UEFI variable directly.
+# Below are the offsets defined in UEFI which is 4(fixed header) plus the offset
+# of the variable within the BfModes struct. These offsets are not supposed to
+# change in order to be backward compatible with previous releases.
+#   VARIABLE(Name)       OFFSET(Byte)  SIZE(Byte)
+#   INTERNAL_CPU_MODEL        4            1
+#   HOST_PRIV_LEVEL           5            1
+#   NIC_MODE                  6            1
+#
+sys_cfg_dump_bf_modes()
```

Comment:
The sys_cfg_xxx() is intended for UEFI sys config operations, while this one is not.
Can we use a different name to avoid confusion, such as sysfs_dump_bf_modes()?

Reply:
Yeah I'm happy to rename, but does this change now that we have `SYS_IN_CPU_MODEL`, `MFG_HOST_PRIV_MODE`, and `MFG_PLAT_MODE` as part of the `ARM_CFG_SUBTYPE_SYS` platform config: https://confluence.nvidia.com/display/SW/BlueField+Platform+Configuration+Blob+Format

@kblaiech thoughts?

Reply:
Changed to sysfs_dump_bf_modes and sysfs_dump_pass_settings